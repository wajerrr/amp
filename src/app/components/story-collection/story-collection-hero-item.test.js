import React from 'react';
import renderer from 'react-test-renderer';
import StoryCollectionHeroItem from './story-collection-hero-item';

describe('StoryCollectionHeroItem', () => {
  it('should match snapshot without className', () => {
    const tree = renderer
      .create(
        <StoryCollectionHeroItem
          headline="Who has the right to judge Americans?"
          url="https://www.economist.com/the-economist-explains/2018/05/15/who-has-the-right-to-judge-americans"
          image={{
            width: 1280,
            height: 720,
            url: {
              canonical:
                'https://www.economist.com/sites/default/files/20180512_BLP509.jpg',
            },
          }}
          subheadline="The Economist explains"
          channelUrl="https://www.economist.com/help/about-us"
          channelHeadline="About us"
          description="A minor fraud case provokes a Supreme Court challenge to the expanding power of administrative agencies in government"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with className', () => {
    const tree = renderer
      .create(
        <StoryCollectionHeroItem
          className="class"
          headline="Who has the right to judge Americans?"
          url="https://www.economist.com/the-economist-explains/2018/05/15/who-has-the-right-to-judge-americans"
          image={{
            width: 1280,
            height: 720,
            url: {
              canonical:
                'https://www.economist.com/sites/default/files/20180512_BLP509.jpg',
            },
          }}
          subheadline="The Economist explains"
          channelUrl="https://www.economist.com/help/about-us"
          channelHeadline="About us"
          description="A minor fraud case provokes a Supreme Court challenge to the expanding power of administrative agencies in government"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
