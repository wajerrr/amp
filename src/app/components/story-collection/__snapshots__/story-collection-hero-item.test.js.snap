// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryCollectionHeroItem should match snapshot with className 1`] = `
<li
  className="class"
>
  <a
    className="css-xy0ws6-StyledLinkWrapper e1omscfa3"
    href="https://www.economist.com/the-economist-explains/2018/05/15/who-has-the-right-to-judge-americans"
  >
    <amp-img
      alt="About us"
      height={720}
      layout="responsive"
      sizes="(min-width: 1000px) calc(1000px -  2 * 0.5em), calc(100vw - 2 * 0.5em)"
      src="https://www.economist.com/sites/default/files/20180512_BLP509.jpg"
      srcset="https://www.economist.com/sites/default/files/imagecache/200-width/sites/default/files/20180512_BLP509.jpg 200w,
https://www.economist.com/sites/default/files/imagecache/300-width/sites/default/files/20180512_BLP509.jpg 300w,
https://www.economist.com/sites/default/files/imagecache/400-width/sites/default/files/20180512_BLP509.jpg 400w,
https://www.economist.com/sites/default/files/imagecache/640-width/sites/default/files/20180512_BLP509.jpg 640w,
https://www.economist.com/sites/default/files/imagecache/800-width/sites/default/files/20180512_BLP509.jpg 800w,
https://www.economist.com/sites/default/files/imagecache/1000-width/sites/default/files/20180512_BLP509.jpg 1000w,
https://www.economist.com/sites/default/files/imagecache/1200-width/sites/default/files/20180512_BLP509.jpg 1200w,
https://www.economist.com/sites/default/files/imagecache/1280-width/sites/default/files/20180512_BLP509.jpg 1280w,
https://www.economist.com/sites/default/files/imagecache/1600-width/sites/default/files/20180512_BLP509.jpg 1600w"
      width={1280}
    />
    <div
      className="css-1aepw6l-StyledHeroItemTextContainer e1omscfa2"
    >
      <h3
        className="css-m4ovun-StyledHeader e10thz6k2"
      >
        <span
          className="css-jzz7eb-StyledSubHeadline-headlineStyles e10thz6k1"
        >
          The Economist explains
        </span>
        <span
          className="css-1xpn6sh-StyledHeadline-subHeadlineStyles e10thz6k0"
        >
          Who has the right to judge Americans?
        </span>
      </h3>
      <p
        className="css-1jkfpje-StyledDescription e1omscfa0"
      >
        A minor fraud case provokes a Supreme Court challenge to the expanding power of administrative agencies in government
      </p>
    </div>
  </a>
  <a
    className="e1omscfa1 css-1gtc8lk-StyledLink-StyledSectionLink-StyledSectionLink e1pwg3wv0"
    href="https://www.economist.com/help/about-us"
  >
    About us
    <span
      className={undefined}
    >
      <svg
        version="1.1"
        viewBox="0 0 48 48"
        x="0px"
        xmlSpace="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <polygon
          points="17.6,14.8 20.4,12 32.4,24 20.4,36 17.6,33.2 26.7,24 	"
        />
      </svg>
    </span>
  </a>
</li>
`;

exports[`StoryCollectionHeroItem should match snapshot without className 1`] = `
<li
  className=""
>
  <a
    className="css-xy0ws6-StyledLinkWrapper e1omscfa3"
    href="https://www.economist.com/the-economist-explains/2018/05/15/who-has-the-right-to-judge-americans"
  >
    <amp-img
      alt="About us"
      height={720}
      layout="responsive"
      sizes="(min-width: 1000px) calc(1000px -  2 * 0.5em), calc(100vw - 2 * 0.5em)"
      src="https://www.economist.com/sites/default/files/20180512_BLP509.jpg"
      srcset="https://www.economist.com/sites/default/files/imagecache/200-width/sites/default/files/20180512_BLP509.jpg 200w,
https://www.economist.com/sites/default/files/imagecache/300-width/sites/default/files/20180512_BLP509.jpg 300w,
https://www.economist.com/sites/default/files/imagecache/400-width/sites/default/files/20180512_BLP509.jpg 400w,
https://www.economist.com/sites/default/files/imagecache/640-width/sites/default/files/20180512_BLP509.jpg 640w,
https://www.economist.com/sites/default/files/imagecache/800-width/sites/default/files/20180512_BLP509.jpg 800w,
https://www.economist.com/sites/default/files/imagecache/1000-width/sites/default/files/20180512_BLP509.jpg 1000w,
https://www.economist.com/sites/default/files/imagecache/1200-width/sites/default/files/20180512_BLP509.jpg 1200w,
https://www.economist.com/sites/default/files/imagecache/1280-width/sites/default/files/20180512_BLP509.jpg 1280w,
https://www.economist.com/sites/default/files/imagecache/1600-width/sites/default/files/20180512_BLP509.jpg 1600w"
      width={1280}
    />
    <div
      className="css-1aepw6l-StyledHeroItemTextContainer e1omscfa2"
    >
      <h3
        className="css-m4ovun-StyledHeader e10thz6k2"
      >
        <span
          className="css-jzz7eb-StyledSubHeadline-headlineStyles e10thz6k1"
        >
          The Economist explains
        </span>
        <span
          className="css-1xpn6sh-StyledHeadline-subHeadlineStyles e10thz6k0"
        >
          Who has the right to judge Americans?
        </span>
      </h3>
      <p
        className="css-1jkfpje-StyledDescription e1omscfa0"
      >
        A minor fraud case provokes a Supreme Court challenge to the expanding power of administrative agencies in government
      </p>
    </div>
  </a>
  <a
    className="e1omscfa1 css-1gtc8lk-StyledLink-StyledSectionLink-StyledSectionLink e1pwg3wv0"
    href="https://www.economist.com/help/about-us"
  >
    About us
    <span
      className={undefined}
    >
      <svg
        version="1.1"
        viewBox="0 0 48 48"
        x="0px"
        xmlSpace="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <polygon
          points="17.6,14.8 20.4,12 32.4,24 20.4,36 17.6,33.2 26.7,24 	"
        />
      </svg>
    </span>
  </a>
</li>
`;
